### ISSUE TITLE
Printed kappa value inconsistency between train() and confusionMatrix()

### ISSUE THREAD

**Issue by isabelle981** [topepo/caret#1006]
I analyzed my data using "rf" for method and "Kappa" for metric. I got dramatically different Kappa values in the train() output and the confusionMatrix output. Please see excerpt below. Does anyone have a clue about this? Is this a bug of the program? Thank you!

```
> cRF_out <- train(x=predictors, y=factor(outcome), method="rf", metric="Kappa")
> print(out)

Random Forest 

3975 samples
 173 predictor
   2 classes: 'FALSE', 'TRUE' 

No pre-processing
Resampling: Bootstrapped (25 reps) 
Summary of sample sizes: 3975, 3975, 3975, 3975, 3975, 3975, ... 
Resampling results across tuning parameters:

  mtry  Accuracy   Kappa      
    2   0.8824385  0.001617912
   87   0.8835390  0.082229864
  173   0.8834523  0.091070343

Kappa was used to select the optimal model using the largest value.
The final value used for the model was mtry = 173.

> confusionMatrix(predict(cRF_out), factor(outcome))

Confusion Matrix and Statistics

          Reference
Prediction FALSE TRUE
     FALSE  3509    0
     TRUE      0  466
                                     
               Accuracy : 1          
                 95% CI : (0.9991, 1)
    No Information Rate : 0.8828     
    P-Value [Acc > NIR] : < 2.2e-16  
                                     
                  Kappa : 1          
 Mcnemar's Test P-Value : NA         
                                     
            Sensitivity : 1.0000     
            Specificity : 1.0000     
         Pos Pred Value : 1.0000     
         Neg Pred Value : 1.0000     
             Prevalence : 0.8828     
         Detection Rate : 0.8828     
   Detection Prevalence : 0.8828     
      Balanced Accuracy : 1.0000     
                                     
       'Positive' Class : FALSE   
```

**Comment by topepo** [comment]
There are two reasons that those are not the same: 

1. The one reported by `train` is the average of the 25 Kappa statistics created during resampling

2. The results when you re-predict the training set and compute one Kappa value are well-known to reflect overfitting. With random forest, if you do this and don't get a perfect Kappa, you did something wrong.
